src/core/controller/models/updateApiConfigurationProto.ts

```diff
@@ -15,29 +15,48 @@ export async function updateApiConfigurationProto(
 	controller: Controller,
 	request: UpdateApiConfigurationRequest,
 ): Promise<Empty> {
+	const startTime = performance.now()
+	console.log("[PERF] updateApiConfigurationProto: Starting provider switch")
```

Comment:
Purposely leaving so many console logs here so that @pashpashpash and @celestial-vault can run this on his machine and verify if this works for him too.

Reply:
Once people accept this approach I will remove most of them.